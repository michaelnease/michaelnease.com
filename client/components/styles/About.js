import styled from "styled-components";

export const AboutStyled = styled.section`
  height: 80rem;
  position: relative;
  background: url("/static/images/bio/bio.gif") no-repeat center center;
  background-size: cover;
  border-bottom: 10px solid ${props => props.theme.colors.black};
  @media (max-width: 700px) {
    height: 100rem;
  }
`;

export const Text = styled.div`
  width: 60rem;
  p {
    text-align: justify;
    font-weight: 300;
    font-size: 1.7rem;
    margin-bottom: 20px;
  }
  @media (max-width: 700px) {
    padding: 3rem;
  }
`;

export const Rex = styled.img`
  box-shadow: ${props => props.theme.effects.bs};
  border: 1px solid ${props => props.theme.effects.black};
  float: left;
  width: 20rem;
  margin: ${props => props.theme.spacing.margin};
  position: relative;
  top: 10px;
`;
