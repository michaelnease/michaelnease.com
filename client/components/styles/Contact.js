import styled from "styled-components";

export const Form = styled.form`
  display: flex;
  position: relative;
  align-content: center;
  justify-content: center;
  background-color: ${props => props.theme.colors.darkGray};
  border-bottom: 10px solid ${props => props.theme.colors.black};
  padding: 3rem;
`;

export const Inputs = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 80rem;
  grid-gap: 1.5rem;

  input,
  textarea {
    border: none;
    background-color: ${props => props.theme.colors.gray};
    padding-left: 3rem;
    font-size: 1.5rem;
    box-shadow: ${props => props.theme.effects.bs};
  }

  input {
    height: 5rem;
  }

  textarea {
    padding: 3rem;
    grid-column: span 2;
  }
`;
