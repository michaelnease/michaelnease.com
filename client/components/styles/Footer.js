import styled from "styled-components";

export const Foot = styled.footer`
  padding: 3rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0 10rem;
  p {
    color: ${props => props.theme.colors.gray};
  }

  @media (max-width: 700px) {
    padding: 1rem;
    margin: 0;
  }
`;
