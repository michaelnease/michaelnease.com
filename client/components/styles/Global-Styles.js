import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  @font-face {
    font-family: 'Roboto';
    src: url('/static/fonts/Roboto-Regular.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
  }

  @font-face {
    font-family: 'Roboto';
    src: url('/static/fonts/Roboto-Light.ttf') format('truetype');
    font-weight: 300;
    font-style: normal;
  }

  html {
    box-sizing: border-box;
    font-size: 10px;
  }

  *, *:before, *:after {
    box-sizing: inherit;
  }

  body, h1, h2, h3, h4, h5, h6, p, ol, ul {
    margin: 0;
    padding: 0;
    font-weight: normal;
  }

  ol, ul {
    list-style: none;
  }

  img {
    max-width: 100%;
    height: auto;
  }

  body {
    padding: 0;
    margin: 0;
    font-size: 1.5rem;
    line-height: 2;
    font-family: "Roboto", Arial, Helvetica, sans-serif;
  }

  a {
    text-decoration: none;
  }
  
  button { font-family: "Roboto", Arial, Helvetica, sans-serif; }

  button:focus {
    outline:0;
  }

  textarea:focus, input:focus{
    outline: none;
  }
`;

export default GlobalStyles;
