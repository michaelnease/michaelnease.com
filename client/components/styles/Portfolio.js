import styled from "styled-components";
import Image from "../Image";

export const PortfolioStyled = styled.section`
  position: relative;
  border-bottom: 10px solid ${props => props.theme.colors.black};
`;

export const Work = styled.section`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 20px;
  width: 80rem;
  margin: 10rem 0;
  @media (max-width: 700px) {
    padding: 0 1rem;
  }
`;

export const PortfolioImage = styled(Image)`
  grid-column: span ${props => props.span};
  border: 2px solid ${props => props.theme.colors.black};
  box-shadow: ${props => props.theme.effects.bs};
`;
