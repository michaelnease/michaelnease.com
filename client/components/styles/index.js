export * from "./Header";
export * from "./Quotes";
export * from "./About";
export * from "./Technology";
export * from "./Portfolio";
export * from "./Contact";
export * from "./Footer";
export * from "./Send-Message-Button";

export * from "./Texture";
