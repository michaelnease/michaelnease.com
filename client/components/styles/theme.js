export const theme = {
  colors: {
    black: "#141414",
    white: "#fff",
    gray: "#e0e0e0",
    darkGray: "#a9a9a9"
  },
  spacing: {
    margin: "15px",
    padding: "15px"
  },
  effects: {
    bs: "-5px 5px 5px 0px rgba(0, 0, 0, 0.2)"
  },
  bar: {
    height: "60px"
  }
};
