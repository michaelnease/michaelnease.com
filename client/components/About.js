import { AboutStyled, Text, Rex, Texture, Link } from "./styles";

const About = () => {
  return (
    <AboutStyled>
      <Link name="about" />
      <Texture>
        <Text>
          <p>
            I am a Los Angeles based Front-End Software Engineer with over 10
            years of development experience. The latter comes from working on
            enterprise websites which actively receive more than 2 million
            unique visitors a day. My duties include development of new features
            and components, active maintenance, code refactoring, as well as
            unit testing.
          </p>
          <Rex src="/static/images/bio/rex.jpg" />
          <p>
            If I am not writing code, you will probably find me walking my
            üêïnamed Rex through Griffith Park or surfing in Redondo Beach. I
            have recently started listening to tech talks at breweries - what
            could be more enjoyable than learning new technologies while having
            a beer?
          </p>
          <p>
            Below is a small sample of my work. All examples were taken from PSD
            to code. Technology used include React/Redux, HTML, CSS, PHP, and
            Magento.
          </p>
        </Text>
      </Texture>
    </AboutStyled>
  );
};

export default About;
