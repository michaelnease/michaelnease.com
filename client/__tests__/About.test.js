import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { render } from "react-testing-library";
import "jest-styled-components";
import { ThemeProvider } from "styled-components";

import { theme } from "../components/styles/theme";
import About from "../components/About";

configure({ adapter: new Adapter() });

describe("<About />", () => {
  const wrapper = shallow(<About />);
  const imagePath = "/static/images/bio/rex.jpg";

  it("Should render three <p /> elements", () => {
    expect(wrapper.find("p").length).toBe(3);
  });

  it("Should render one <Rex /> element", () => {
    expect(wrapper.find("Rex").length).toBe(1);
  });

  it("Should render <Rex /> with src", () => {
    expect(wrapper.find("Rex").prop("src")).toEqual(imagePath);
  });

  it("Should match Snapshot", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <About />
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
