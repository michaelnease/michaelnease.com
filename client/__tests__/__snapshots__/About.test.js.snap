// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> Should match Snapshot 1`] = `
<DocumentFragment>
  .c1 {
  position: absolute;
  top: -60px;
}

.c0 {
  height: 80rem;
  position: relative;
  background: url("/static/images/bio/bio.gif") no-repeat center center;
  background-size: cover;
  border-bottom: 10px solid #141414;
}

.c3 {
  width: 60rem;
}

.c3 p {
  text-align: justify;
  font-weight: 300;
  font-size: 1.7rem;
  margin-bottom: 20px;
}

.c4 {
  box-shadow: -5px 5px 5px 0px rgba(0,0,0,0.2);
  border: 1px solid;
  float: left;
  width: 20rem;
  margin: 15px;
  position: relative;
  top: 10px;
}

.c2 {
  background-image: url("/static/images/common/texture.png");
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (max-width:700px) {
  .c0 {
    height: 100rem;
  }
}

@media (max-width:700px) {
  .c3 {
    padding: 3rem;
  }
}

<section
    class="c0"
  >
    <a
      class="c1"
      name="about"
    />
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <p>
          I am a Los Angeles based Front-End Software Engineer with over 10 years of development experience. The latter comes from working on enterprise websites which actively receive more than 2 million unique visitors a day. My duties include development of new features and components, active maintenance, code refactoring, as well as unit testing.
        </p>
        <img
          class="c4"
          src="/static/images/bio/rex.jpg"
        />
        <p>
          If I am not writing code, you will probably find me walking my üêïnamed Rex through Griffith Park or surfing in Redondo Beach. I have recently started listening to tech talks at breweries - what could be more enjoyable than learning new technologies while having a beer?
        </p>
        <p>
          Below is a small sample of my work. All examples were taken from PSD to code. Technology used include React/Redux, HTML, CSS, PHP, and Magento.
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
